pkg/hydrate/controller.go

```diff
@@ -115,13 +115,32 @@ func (h *Hydrator) runHydrate(sourceCommit, syncDir string) HydrationError {
 	if err := kustomizeBuild(syncDir, dest, true); err != nil {
 		return err
 	}
+
+	changed, commit, err := h.checkCommitChanged(sourceCommit)
+	if err != nil {
+		return NewTransientError(err)
+	} else if changed && commit != "" {
+		return NewTransientError(fmt.Errorf("source commit changed while running Kustomize build, was %s, now %s. It will be retried in the next sync", sourceCommit, commit))
+	}
+
 	if err := updateSymlink(h.HydratedRoot.OSPath(), h.HydratedLink, newHydratedDir.OSPath()); err != nil {
 		return NewInternalError(errors.Wrapf(err, "unable to update the symbolic link to %s", newHydratedDir.OSPath()))
 	}
 	klog.Infof("Successfully rendered %s for commit %s", syncDir, sourceCommit)
 	return nil
 }
 
+// checkCommitChange returns true and the updated commit if the computed commit
+// is different from the given commit, or returns error if found during computation
+func (h *Hydrator) checkCommitChanged(commit string) (bool, string, error) {
```

Comment:
The boolean return value seems unnecessary here, I'd suggest just returning the commit `string` and doing the comparison in the caller. e.g.
```golang
	commit, err := h.checkCommitChanged() // potentially rename the function
	if err != nil {
		return NewTransientError(err)
	} else if commit != sourceCommit {
		return NewTransientError(fmt.Errorf("source commit changed while running Kustomize build, was %s, now %s. It will be retried in the next sync", sourceCommit, commit))
	}
```

Reply:
Simplified the return values